WordPress/gutenberg
# Refactor experimental dropdown menu usages to latest version

## What?


Refactor existing usages of `DropdownMenuV2` to the newest version of the component. The plan is to delete that older version after this PR is merged (#55626)

## Why?


There is a new version of the experimental `DropdownMenu` "v2" component supposed to replace the previous experimental version.

## How?


Most refactors were a 1:1 swap between the old and the new version of the component. A few exceptions:

- SubMenu and SubMenu Triggers: with the new version, we can just reuse `DropdownMenu` and `DropdownMenuItem` and they will work in nested menus automatically
- Many instances of `DropdownMenuItem` were actually fulfilling a radio item role (without the need for "de-selecting" an item by clicking it again), and therefore I refactored them to the `DropdownMenuRadioItem` component (which removes the need to explicitly add radio semantics and show a check icon)
- As per the latest design specs, I removed decorative icons from radio/checkbox menu items
- I wrapped the extra info in the `suffix` in `<span aria-hidden="true" />` to hide from assistive technology
- Refactored and simplified the code around IN / NOT IN operators

## Testing Instructions





- Enable the "New admin views" Gutenberg experiment
- In the site editor, select "Pages" and then "Manage all pages"
- Check that the actions dropdown (under the "View" button) and column header dropdown work as on trunk

## Screenshots or screencast 

![Screenshot 2023-10-25 at 17 58 22](https://github.com/WordPress/gutenberg/assets/1083581/7986df78-63c1-4ef5-9ab0-68c4106925fa)

![Screenshot 2023-10-25 at 17 59 04](https://github.com/WordPress/gutenberg/assets/1083581/4748224d-8c87-4a7a-9e62-60d046cb1046)
